connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockSetLocalPlayerAsInitializedTranslator.java

```diff
@@ -48,6 +50,16 @@ public void translate(SetLocalPlayerAsInitializedPacket packet, GeyserSession se
                         entity.sendPlayer(session);
                     }
                 }
+
+                // Send Skulls
+                for (PlayerEntity entity : session.getSkullCache().values()) {
+                    entity.spawnEntity(session);
+
+                    SkinUtils.requestAndHandleSkinAndCape(entity, session, (skinAndCape) -> session.getConnector().getGeneralThreadPool().schedule(() -> {
+                        entity.getMetadata().getFlags().setFlag(EntityFlag.INVISIBLE, false);
+                        entity.updateBedrockMetadata(session);
+                    }, 2, TimeUnit.SECONDS));
```

Comment:
Would recommend scheduling every Skull in one schedule.
Spawn every skull, make a schedule and paste every update and skin / cape? request in that one schedule.